pkg/storage/minio/getter.go

```diff
@@ -49,7 +49,13 @@ func (v *storageImpl) Zip(ctx context.Context, module, vsn string) (io.ReadClose
 	const op errors.Op = "minio.Zip"
 	sp, ctx := opentracing.StartSpanFromContext(ctx, "storage.minio.Zip")
 	defer sp.Finish()
+
 	zipPath := fmt.Sprintf("%s/source.zip", v.versionLocation(module, vsn))
+	_, err := v.minioClient.StatObject(v.bucketName, zipPath, minio.StatObjectOptions{})
```

Comment:
is there no explicit check for NotFound? What if this was a connection error or something?

Reply:
yes, but at the point you read the blob. you can see that at the other places here. but in case of zip we don't want to read it, we need client to read it